test/common/network/dns_impl_test.cc

```diff
@@ -859,5 +863,60 @@ TEST(DnsImplUnitTest, PendingTimerEnable) {
                                       }));
 }
 
+class DnsImplAresFlagsForTcpTest : public DnsImplTest {
+protected:
+  bool tcp_only() const override { return false; }
+  bool use_tcp_for_dns_lookups() const override { return true; }
+};
+
+// Parameterize the DNS test server socket address.
+INSTANTIATE_TEST_SUITE_P(IpVersions, DnsImplAresFlagsForTcpTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// Validate that c_ares flag `ARES_FLAG_USEVC` is set when boolean property
+// `use_tcp_for_dns_lookups` is enabled.
+TEST_P(DnsImplAresFlagsForTcpTest, TcpLookupsEnabled) {
+  server_->addCName("root.cnam.domain", "result.cname.domain");
```

Comment:
Can you elaborate on how these tests are hermetic? I assume they don't rely on any actual DNS reachability.

Reply:
@htuch in this test set up we spin up dns server at loopback address: https://github.com/envoyproxy/envoy/blob/master/test/common/network/dns_impl_test.cc#L411